### ISSUE TITLE
Map transmissionIndices bug

### ISSUE THREAD

**Issue by jameshadfield** [nextstrain/auspice#547]
Sometimes this error occurs:
```
mapHelpersLatLong.js?a8f2:402 Uncaught TypeError: Cannot read
   property 'forEach' of undefined
       at eval (mapHelpersLatLong.js?a8f2:402)
       at Array.forEach (<anonymous>)
       at eval (mapHelpersLatLong.js?a8f2:389)
       at Array.forEach (<anonymous>)
       at updateTransmissionDataColAndVis (mapHelpersLatLong.js?a8f2:387)
       at updateDemeAndTransmissionDataColAndVis
   (mapHelpersLatLong.js?a8f2:428)
       at Map.maybeUpdateDemesAndTransmissions (map.js?e5a9:312)
       at Map.componentWillReceiveProps (map.js?e5a9:105)
       at callComponentWillReceiveProps
   (react-dom.development.js?cada:6389)
       at updateClassInstance (react-dom.development.js?cada:6575)
   ----
```
Caused by `transmissionIndices[id]` being undefined.

A try-catch fix was implemented in 1.18.1 (https://github.com/nextstrain/auspice/commit/ee4bca6a58a8175f7fdd1cfaf60dd1796fc818b4) however the underlying issue persists.

**Comment by jameshadfield** [comment]
@trvrb are you familiar with colin's code here?

**Comment by trvrb** [comment]
Can you tell if this was introduced recently? Or is persistent? And is there a reliable way to reproduce?

**Comment by jameshadfield** [comment]
JSONs in the slack channel. May be a JSON thing, i've never seen it before.

**Comment by rneher** [comment]
fundamentally, the issue here is to decide what to do/how to fail when the meta.json doesn't contain coordinates. in the case above, this happened because an ancestral state was inferred that wasn't even in the country list (I think we pinned the ancestral state to vietnam, but vietnam wasn't sampled and hence excluded from the `geo` fields in the meta json).

**Comment by jameshadfield** [comment]
Closed in #608
